validator/src/main.rs

```diff
@@ -1754,6 +1754,8 @@ pub fn main() {
         BlockProductionMethod
     )
     .unwrap_or_default();
+    validator_config.enable_block_production_forwarding =
+        matches.is_present("enable_block_production_forwarding");
```

Comment:
wdyt of inferring this from the corresponding swqos flags being passed instead of an explicit flag? we can skip the config complexity and ensuing support burden

Reply:
I'm not familiar with how people do the staked-node-overrides.

It seems like it's possible that as a validator I have a side deal to override their stake so that they get more bandwidth when I'm leader, without me having to forward.

Not sure anybody is using it like this, but making the forward enabling based on staked-node-overrides makes that impossible.

Reply:
I would look at it from the perspective that the described opportunity exists right now only due to a confluence of circumstances like the new version of the scheduler and the current implementation of SWQoS.

However, to support such users, you could do something like this:
```
arg "disable_block_production_forwarding" requires "staked-nodes-overrides"

enable_block_production_forwarding = is_present "staked-nodes-overrides" && !is_present "disable_block_production_forwarding"

```

Then, 99% of validators won't need to do anything. Forwarding will be disabled by default. For the 99% of validators who, I assume, use the main case of landing traffic in the cluster, they also won’t need to configure anything additional. And only those who land traffic only on their own will need to set an additional flag.

Reply:
THink that makes sense, gives us best of both worlds: ef364a197f840a2dc46e3ce481dc30f28a2f5ac6